modules/tinymce/src/themes/silver/main/ts/ui/context/ContextToolbarLookup.ts

```diff
@@ -17,15 +17,41 @@ type MatchResult = { contextToolbars: ContextTypes[]; contextForms: ContextTypes
 
 const matchTargetWith = (elem: Element, candidates: ContextTypes[]): MatchResult => {
   const ctxs = Arr.filter(candidates, (toolbarApi) => toolbarApi.predicate(elem.dom()));
+  // TODO: change Arr.partition's types or something to allow pass and fail to have different types
```

Comment:
Worth noting this won't ever work with TS in its current state, as you can only add a type guard for the `pass` case. So you're probably just better casting them as needed.

Reply:
Point. Sigh. Changed my comment for now.